socorro/external/postgresql/connection_context.py

```diff
@@ -68,21 +81,41 @@ def __init__(self, config, local_config=None):
             local_config - this is the namespace within the complete config
                            where the actual database parameters are found"""
         super(ConnectionContext, self).__init__()
+
         self.config = config
+
         if local_config is None:
             local_config = config
+
+        if local_config['database_url']:
+            self.database_url = local_config['database_url']
+            url = urlparse.urlparse(local_config['database_url'])
+            local_config['database_username'] = url.username
+            local_config['database_password'] = url.password
+            local_config['database_hostname'] = url.hostname
+            local_config['database_port'] = url.port
+            local_config['database_name'] = url.path[1:]
+
+            self.db_kwargs = {
+                'host': url.hostname or 'localhost',
+                'user': url.username or os.environ.get('USER', 'test'),
```

Comment:
Shouldn't all of these defaults already be built into the default `database_url`? In particular that user field looks like something that can cause debugging confusion some day. 

Why not do something like this:
```
self.db_kwargs = {}
if url.hostname:
    self.db_kwargs['host'] = url.hostname
...
```
Then we're leaving the defaultism up to `psycopg2.connect()` instead.